### PR TITLE
[#496] Distribute app without Firebase integration

### PR DESCRIPTION
**Pull request by lydiasama**
- Close https://github.com/nimblehq/android-templates/issues/496

## What happened 👀

Remove Firebase integration from project

## Insight 📝

- Codemagic can distribute the app to Firebase App Distribution without integrating the Firebase to the project.
- After merging this PR, the `google-services.json` in Github Secrets will be deleted.

Big thanks to @luongvo for noticing this and the POC branch. 🙇🏻‍♀️ 

## Proof Of Work 📹

Github Action to deploy on Firebase result:

https://github.com/nimblehq/android-templates/actions/runs/5712267786/job/15475425029

Codemagic to deploy on Firebase result:

Screenshot 2566-08-02 at 4 54 16 PM

### CODE REVIEW

File: codemagic.yaml [hunk 1]
```diff
@@ -10,13 +10,6 @@ definitions:
       cache_paths:
         - $HOME/.gradle/caches
   scripts:
-    - &set_up_google_services_files_from_environment_variables
-      name: Set up google-services.json on template-compose
-      script: |
-        mkdir -p template-compose/app/src/production
-        echo $GOOGLE_SERVICES_JSON > template-compose/app/src/production/google-services.json
-        mkdir -p template-compose/app/src/staging
-        echo $GOOGLE_SERVICES_JSON_STAGING > template-compose/app/src/staging/google-services.json
```
Comment (luongvo) [inline codemagic.yaml:19]: Should we also clean up all unnecessary steps in https://github.com/search?q=repo%3Animblehq%2Fandroid-templates+GOOGLE_SERVICES_JSON&type=code

Comment (luongvo) [inline codemagic.yaml:19]: Should we also clean up all unnecessary steps in https://github.com/search?q=repo%3Animblehq%2Fandroid-templates+GOOGLE_SERVICES_JSON&type=code

**Comment by lydiasama** [comment]
All updated. @luongvo  🙏🏻

**Review by luongvo — COMMENTED** [review]
There are still 1 more place to temporary remove:
- https://github.com/nimblehq/android-templates/blob/3f234ab29313997961b4526956b877b3ee0e2a12/.github/workflows/review_pull_request.yml#L50-L58

**Comment by lydiasama** [comment]
@luongvo I just recalled why I didn't remove it. If we use this `github action workflow` from `.cicdtemplate` in any project to distribute the app to the Firebase, that means we need to set up the Firebase to the project. So, we recommended saving the `google-service.json` file in the Github Secrets. 🤔  So IMHO, we don't need to remove that step to download the file in the `.cicdtempleate` folder. What do you think?

**Comment by luongvo** [comment]
> If we use this github action workflow from .cicdtemplate in any project to distribute the app to the Firebase, that means we need to set up the Firebase to the project. 

Yes, we need to set up a new Firebase app on the Firebase console for distribution, but we **do not need to add Firebase to the app**. Hence, no google-services is required.

> And we recommended saving the google-service.json file in the Github Secrets. 🤔 So IMHO, we don't need to remove that step to download the file in the .cicdtempleate folder.

We can keep that variable on GitHub Secrets for later usage. For now, we do not use it to execute any commands or to build the app, so I recommend removing the step to download it to avoid confusion. 

@lydiasama Since you removed from both the `root workflows` and the `.cicdtemplate`, we should also remove it in https://github.com/nimblehq/android-templates/blob/3f234ab29313997961b4526956b877b3ee0e2a12/.github/workflows/review_pull_request.yml#L50-L58 🙏

**Comment by luongvo** [comment]
@lydiasama @ryan-conway can we resolve and merge this PR asap? I'm working on KMM Templates and to reuse the Android app module, I need to have a successful build without this config 🙏

**Review by ryan-conway — APPROVED** [review]

**Review by luongvo — APPROVED** [review]

**Review by AVI5HEK — APPROVED** [review]

**Pull request merged**